pkg/handlers/generic/mutation/imageregistries/credentials/inject.go

```diff
@@ -151,6 +152,33 @@ func (h *imageRegistriesPatchHandler) Mutate(
 		return generateErr
 	}
 
+	credentialsSecret, generateErr := generateCredentialsSecret(
```

Comment:
`generateCredentialsSecret` has side effects. it returns `nil,nil` if `registriesWithOptionCredentials` is empty. https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/main/pkg/handlers/generic/mutation/imageregistries/credentials/credentials_secret.go#L57.
```
if len(configs) == 0 {
		return nil, nil
	}
```
This will break this mutator. 
It think we need to refactor that function to move that condition outside.